src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3652,11 +3652,12 @@ void ViewProviderSketch::InitItemsSizes()
     int defaultFontSize = QApplication::fontMetrics().height();
     int ldpi = QApplication::desktop()->logicalDpiX();
     float virtualdpi = 96.;
-    float k = virtualdpi/ldpi;
+    float QtPixelRatio = virtualdpi/ldpi;
+    float coinFontPixelRatio = QtPixelRatio; // this is not absolute exactly, but the ratio is correct
     float view3D_factor = 1.25; // View3D area has worse readability, so let's increase a little
     ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/View");
-    coinFontSize = hGrp->GetInt("EditSketcherFontSize", defaultFontSize * k * k *view3D_factor );
-    constraintIconSize = coinFontSize / k;
+    coinFontSize = hGrp->GetInt("EditSketcherFontSize", defaultFontSize * QtPixelRatio * coinFontPixelRatio *view3D_factor );

```

Comment:
So, basically, you get the default font size in pixels and interpret it as the font size in points.
Then you multiply it by QtPixelRatio to get font size in pixels.
The you interpret it as the font size in points and multiply it by coinFontPixelRatio to get font size in pixels.

It doesn't make much sense.